File: 백준/Bronze/12813. 이진수 연산/이진수 연산.py
A = int(input(), 2)
B = int(input(), 2)

fill_num = 100000
mask = 2 ** fill_num - 1

print(bin(A & B)[2:].zfill(fill_num))
print(bin(A | B)[2:].zfill(fill_num))
print(bin(A ^ B)[2:].zfill(fill_num))
print(bin(A ^ mask)[2:].zfill(fill_num))
print(bin(B ^ mask)[2:].zfill(fill_num))

File: 백준/Bronze/12813. 이진수 연산/이진수 연산.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : 'run/input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const [aStr, bStr] = input;

const a = BigInt('0b' + aStr);
const b = BigInt('0b' + bStr);

const fill = 100000;
const mask = 2n ** 100000n - 1n;

console.log((a & b).toString(2).padStart(fill, '0'));
console.log((a | b).toString(2).padStart(fill, '0'));
console.log((a ^ b).toString(2).padStart(fill, '0'));
console.log((a ^ mask).toString(2).padStart(fill, '0'));
console.log((b ^ mask).toString(2).padStart(fill, '0'));
